src/transformers/utils/deprecation.py

```diff
@@ -0,0 +1,91 @@
+from functools import wraps
+from typing import Optional
+from warnings import warn
+
+import packaging.version
+
+from .. import __version__
+from . import ExplicitEnum
+
+
+class Action(ExplicitEnum):
+    RAISE = "raise"
+    NOTIFY = "notify"
+
+
+def warning(message):
+    warn(message, FutureWarning)
+
+
+def deprecate_kwarg(
+    old_name: str,
+    version: str,
+    new_name: Optional[str] = None,
+    raise_if_ge_version: bool = False,
+    raise_if_both_names: bool = False,
+    add_message: Optional[str] = None,
```

Comment:
nit - could we call this `additional_message`? `add_message` reads as a bool to me (don't feel strongly about this, so happy if you decide to stick with the original)